Bugfix: Get image Thumbnail Admin Action

TobiSW opened the pull request:
- Call getById from Asset instead of Asset\Image to not break class model overrides

## Steps to reproduce
- Overwrite the Pimcore\Model\Asset\Image in your config.yaml models -> class_overrides as described here:
https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Overriding_Models.html
- Upload a new Asset in the DAM and click directly on the uploaded image (No reload)
- getImageThumbnailAction gets the image
- The image is an instance of Pimcore\Model\Asset\Image instead of your own overwritten class

<!--

Before working on a contribution, you must determine on which branch you need to work:
- Bug fix: choose the latest maintenance branch `10.5`
- Feature/Improvement: choose `11.x` 

> All bug fixes merged into the latest maintenance branch are also merged to the current dev branch (`11.x`) on a regular basis.

## Please make sure your PR complies with all of the following points: 
- [ ] Read and accept our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR.
- [ ] Features need to be proper documented in `doc/` 
- [ ] Bugfixes need a short guide how to reproduce them -> target branch is the oldest supported maintenance branch, e.g. `10.5` (see Readme.md for the list of supported versions)
- [ ] Meet all coding standards (see PhpStan actions) 

**Don't submit a PR if it doesn't comply, it'll be closed without a comment!**
-->  
  

## Changes in this pull request  
Change how a image will be loaded in the imageThumbnailAction

## Additional info  
The Pimcore\Model\Asset getById method which is being called in the Pimcore\Bundle\AdminBundle\Controller\Admin\Asset\AssetContoller getImageThumbnailAction creates a new static instance.

https://github.com/pimcore/pimcore/blob/87ec1423edc77b5c78a05709146ecc092c055ce8/models/Asset.php#L250-L261

This will break class model overrides in this case. To avoid this the method should be called from the Asset itself instead of Asset\Image

TobiSW committed:
Bugfix: Get image Thumbnail Admin Action

- Call getById from Asset instead of Asset\Image to not break class model overrides

mcop1 commented:
@TobiSW could you please have a look at the failing tests? Thanks!

blankse commented:
@TobiSW Can you test my fix https://github.com/pimcore/pimcore/pull/13561 ?
This should be a better fix for this.

TobiSW committed:
Bugfix: Get image Thumbnail Admin Action

- Call getById from Asset instead of Asset\Image to not break class model overrides

TobiSW commented:
> @TobiSW Can you test my fix #13561 ? This should be a better fix for this.

thanks very cool. I tested it out, your fix is also working.
I was also already pushing an update , which seems also to work. Which solution do you prefer?

blankse commented:
@TobiSW I prefer my fix.
With your change the Dao::getById() would be called always twice, although the right class is given.
My PR fix also the Document::getById() method.

TobiSW commented:
@blankse okay nvm lets go ahead with your one, thanks for bringing this one in :)
I will close this one and reference to your one https://github.com/pimcore/pimcore/pull/13561.
